### PR TITLE
Support multiple ignore files (address #29)

### PR DESCRIPTION
**Pull request by alvinasa**
This should address #29

### CODE REVIEW

File: tests/rsync/rsync_spec.lua [hunk 1]
```diff
@@ -505,6 +505,104 @@ describe("rsync", function()
...
         end)
+
+        local function setup_with_multiple_ignore_files(code)
+            helpers.write_file(".nvim/rsync.toml", { 'remote_path = "' .. helpers.dest .. '/"', 'ignorefile_paths = [ ".gitignore.1", ".gitignore.2" ]' })
+            helpers.write_file(".gitignore.1", { "local_should_ignore_1.txt", "remote_should_ignore_1.txt" })
+            helpers.write_file(".gitignore.2", { "local_should_ignore_2.txt", "remote_should_ignore_2.txt" })
+            helpers.write_file("local_should_ignore_1.txt", { "this file\nshould not be synced" })
+            helpers.write_file("local_should_ignore_2.txt", { "this file\nshould not be synced" })
+            helpers.write_file("local.c", { "eueueu" })
+            helpers.mkdir_remote("")
```
Comment (OscarCreator) [inline tests/rsync/rsync_spec.lua:516]:
```suggestion
```
Do not think this should be needed.

File: tests/rsync/rsync_spec.lua [hunk 2]
```diff
@@ -505,6 +505,104 @@ describe("rsync", function()
...
+        end)
+
+        local function setup_with_no_ignore_files(code)
+            -- ignorefile_paths is explicitly set to none
+            helpers.write_file(".nvim/rsync.toml", { 'remote_path = "' .. helpers.dest .. '/"', 'ignorefile_paths = []' })
+
+            -- this ignore file is not be honored
+            helpers.write_file(".gitignore", { "local_should_sync.txt" })
+            helpers.write_file("local_should_sync.txt", { "this file\nshould be synced" })
+            helpers.mkdir_remote("")
```
Comment (OscarCreator) [inline tests/rsync/rsync_spec.lua:581]:
```suggestion
```
same here

**Review by OscarCreator — CHANGES_REQUESTED** [review]
Thanks for the PR!
Looks good, added a few  comments.

Also it would be nice if you updated the documentation in README.md and doc/rsync.txt with the new option.

File: lua/rsync/sync.lua [hunk 3]
```diff
@@ -103,7 +105,7 @@ function sync.sync_up(report_error)
 
             vim.fn.jobstop(current_status.job_id)
         end
-        local command = compose_sync_up_command(config_table.project_path, config_table.remote_path)
+        local command = compose_sync_up_command(config_table.project_path, config_table.remote_path, config_table.ignorefile_paths or {".gitignore"})
```
[the remark this answers is not in the document]
Comment (alvinasa) [reply lua/rsync/sync.lua:108]: That's a good point. Updated.

**Comment by alvinasa** [comment]
Thanks @OscarCreator for reviewing!

I also formatted the code with `stylua`. I can run it against a config or another formatter if you have any preference.

**Pull request merged**